Name: Fix soft-lock when paused and unable to move
Author: Adrian Siekierka <kontakt@asie.pl>

--- src.orig/SRC/GAME.PAS	2020-03-16 14:53:28.665824692 +0100
+++ src/SRC/GAME.PAS	2020-03-16 23:22:54.277357503 +0100
@@ -1417,6 +1417,7 @@ procedure GamePlayLoop(boardChanged: boo
 	var
 		exitLoop: boolean;
 		pauseBlink: boolean;
+		unpauseGame: boolean;
 	procedure GameDrawSidebar;
 		begin
 			SidebarClear;
@@ -1535,7 +1536,9 @@ procedure GamePlayLoop(boardChanged: boo
 				if InputKeyPressed = KEY_ESCAPE then
 					GamePromptEndPlay;
 
+				unpauseGame := false;
 				if (InputDeltaX <> 0) or (InputDeltaY <> 0) then begin
+					unpauseGame := true;
 					ElementDefs[Board.Tiles[Board.Stats[0].X + InputDeltaX][Board.Stats[0].Y + InputDeltaY].Element].TouchProc(
 						Board.Stats[0].X + InputDeltaX, Board.Stats[0].Y + InputDeltaY, 0, InputDeltaX, InputDeltaY);
 				end;
@@ -1556,9 +1559,13 @@ procedure GamePlayLoop(boardChanged: boo
 						DrawPlayerSurroundings(Board.Stats[0].X, Board.Stats[0].Y, 0);
 						DrawPlayerSurroundings(Board.Stats[0].X - InputDeltaX, Board.Stats[0].Y - InputDeltaY, 0);
 					end;
+				end;
 
+				if unpauseGame then begin
 					{ Unpause }
 					GamePaused := false;
+					{ The player might be invisible if he hasn't moved in the above code }
+					BoardDrawTile(Board.Stats[0].X, Board.Stats[0].Y);
 					SidebarClearLine(5);
 					CurrentTick := Random(100);
 					CurrentStatTicked := Board.StatCount + 1;
